pkg/apis/core/validation/validation.go

```diff
@@ -3037,73 +3095,18 @@ func validateContainers(containers []core.Container, isInitContainers bool, volu
 	allNames := sets.String{}
 	for i, ctr := range containers {
 		idxPath := fldPath.Index(i)
-		namePath := idxPath.Child("name")
-		volMounts := GetVolumeMountMap(ctr.VolumeMounts)
-		volDevices := GetVolumeDeviceMap(ctr.VolumeDevices)
 
-		if len(ctr.Name) == 0 {
-			allErrs = append(allErrs, field.Required(namePath, ""))
-		} else {
-			allErrs = append(allErrs, ValidateDNS1123Label(ctr.Name, namePath)...)
-		}
 		if allNames.Has(ctr.Name) {
-			allErrs = append(allErrs, field.Duplicate(namePath, ctr.Name))
+			allErrs = append(allErrs, field.Duplicate(idxPath.Child("name"), ctr.Name))
```

Comment:
Do we care if there is a dup name across init/ephemeral/real containers?

Reply:
We do care, but we check for it later to avoid duplicate error messages. I've added comments explaining how the checks work.